### PR TITLE
Update plugin api, reorganize

### LINKED ISSUE #18: inspec-iggy doesn't works with inspec 3.0.0
Try to install inspec-iggy using the following command. It fails.
```
inspec plugin install -v 0.3.0 inspec-iggy
inspec plugin install -v 0.2.0 inspec-iggy
```

Most probably inspec dependency needs to be updated for inspec 3.0.0

### PR DESCRIPTION
**Pull request by clintoncwolfe**
This is a broad PR, with three major goals:

 * Update the plugin API from v1 to v2
 * Organize testing to include several types, and strong unit testing on the plugin
 * Re-organize the plugin layout, including placing the two existing IaaC drivers into parallel directories; namespacing the modules to be under `InspecPlugins::Iggy`, and building out the Rakefile and Gemfile.
 * Added functional tests and verified it works.
 * Locally changed version to 0.3.99, built gem, and verified installation.

Fixes #18 

screen shot 2018-10-18 at 02 56 11

### CODE REVIEW

File: inspec-iggy.gemspec [hunk 1]
```diff
@@ -22,5 +22,5 @@ Gem::Specification.new do |spec|
 
   spec.require_paths = ['lib']
 
-  spec.add_dependency 'inspec', '>=2.2.64', '<3.0.0'
+  spec.add_dependency 'inspec', '>=2.3', '<4.0.0'
```
Comment (mattray) [inline inspec-iggy.gemspec]: @clintoncwolfe I'm guessing <4.0.0 is pretty safe?

File: lib/inspec-iggy/version.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@
 #
 # Copyright:: 2018, Chef Software, Inc <legal@chef.io>
 #
-
-# Iggy version
-module Iggy
-  VERSION = '0.3.0'.freeze
+module InspecPlugins
+  module Iggy
+    VERSION = '0.3.0'.freeze
```
Comment (mattray) [inline lib/inspec-iggy/version.rb]: This can go to 0.4.0

File: lib/inspec-iggy/plugin.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        # functionality.
+        # For example, InSpec will activate this hook when `inspec help` is
+        # executed, so that this plugin's usage message will be included in the help.
+        require 'inspec-iggy/terraform_cli'
+
+        # Having loaded our functionality, return a class that will let the
+        # CLI engine tap into it.
+        InspecPlugins::Iggy::TerraformCliCommand
+      end
+    end
```
Comment (mattray) [inline lib/inspec-iggy/plugin.rb]: Where did the entry point for "inspec cloudformation" go? Can 2 cli_commands be defined here?

Comment (clintoncwolfe) [reply lib/inspec-iggy/plugin.rb]: great catch!  how are the functional tests coming?

File: README.md [hunk 4]
```diff
@@ -164,15 +109,40 @@ Iggy supports AWS CloudFormation templates by mapping the AWS resources to InSpe
...
 
-    $ rspec
+```
+$ inspec plugin install path/to/your/inspec-iggy/lib/inspec-iggy.rb
+```
+
+# Testing Iggy<a name="testing"></a>
+
+
+Unit, Functional, and Integration tests are provided, though more are welcome. Iggy uses the RSpec library for testing, using the `expect('subject').to matcher` syntax.
```
Comment (jerryaldrichiii) [inline README.md]: Are these tests actually provided?

File: inspec-iggy.gemspec (hunk 1 above)
Comment (clintoncwolfe) [reply inspec-iggy.gemspec]: 2.3 introduced the new plugin API.  No relevant breaking changes on 3.0 for iggy, next pin is 4.

File: lib/inspec-iggy/version.rb (hunk 2 above)
Comment (clintoncwolfe) [reply lib/inspec-iggy/version.rb]: We'll make sure it gets bumped after merge to master.

File: lib/inspec-iggy/plugin.rb (hunk 3 above)
Comment (clintoncwolfe) [reply lib/inspec-iggy/plugin.rb]: I've added back in the cloudformation entrypoint (that was a simple oversight).

I've written functional tests.

File: README.md (hunk 4 above)
Comment (clintoncwolfe) [reply README.md]: Yes.  See the `test/` directory.

**Pull request merged**